package/linux-pam/0005-faillock-remove-user-resolution.patch

```diff
@@ -0,0 +1,35 @@
+--- a/modules/pam_faillock/pam_faillock.c
++++ b/modules/pam_faillock/pam_faillock.c
+@@ -374,28 +374,10 @@
+ 		return PAM_IGNORE;
+ 	}
+ 
+-	if ((pwd=pam_modutil_getpwnam(pamh, user)) == NULL) {
+-		if (opts->flags & FAILLOCK_FLAG_AUDIT) {
+-			pam_syslog(pamh, LOG_NOTICE, "User unknown: %s", user);
+-		}
+-		else {
+-			pam_syslog(pamh, LOG_NOTICE, "User unknown");
+-		}
+-		return PAM_IGNORE;
+-	}
+-	opts->user = user;
+-	opts->uid = pwd->pw_uid;
+-
+-	if (pwd->pw_uid == 0) {
+-		opts->is_admin = 1;
+-		return PAM_SUCCESS;
+-	}
+-
+-	if (opts->admin_group && *opts->admin_group) {
+-		opts->is_admin = pam_modutil_user_in_group_uid_nam(pamh,
+-			pwd->pw_uid, opts->admin_group);
+-	}
+-
++	// sikli changes: remove user-name -> uid resolution; hardcode output values:
++	opts->user = user; 	// to open tally file
++	opts->uid = 0;		// uid of tally file
++	opts->is_admin = 0;	// never treat the user as root/admin with specific configuration options (even_deny_root, root_unlock_time)
```

Comment:
1. can you please explain this in a comment?
2. root and admin are two different users. Can you explain this?

Reply:
@dima1308, do you mean comment if this source file or comment for this PR?

Reply:
yes, it is better to add more explanation to the diff file. It is not clear to me how the changes you made solve the problem and how the faillock will work with non-radios/tacacs+ users after this change.
It is better to add much more detailed information. The reason for this is that not all of us are familiar with the logic of the PAM and pam-faillock

Reply:
@dima1308 
on other hand -- patch file deep withing standalone repository isn't best place for long explanations, especially when "not all of us are familiar with the logic of the PAM and pam-faillock code."

really

Reply:
added long comment